packages/core/src/dialects/mssql/query-interface-typescript.ts

```diff
@@ -0,0 +1,95 @@
+      throw new TypeError('Unable to create a savepoint without transaction object!');
+    }
+
+    const connection = transaction.getConnection() as MsSqlConnection;
+    await connection.queue.enqueue(async () => new Promise<void>((resolve, reject) => {
+      // @ts-expect-error -- TODO: remove this when tedious types are fixed
+      connection.saveTransaction(error => (error ? reject(error) : resolve()), options.savepointName);
+    }));
+  }
+
+  async startTransaction(transaction: Transaction, options: StartTransactionOptions): Promise<void> {
+    if (!(transaction instanceof Transaction)) {
+      throw new TypeError('Unable to start a transaction without transaction object!');
+    }
+
+    const connection = transaction.getConnection() as MsSqlConnection;
+    await connection.queue.enqueue(async () => new Promise<void>((resolve, reject) => {
+      connection.beginTransaction(
+        error => (error ? reject(error) : resolve()),
+        options.transactionName,
+        this.#internalQueryInterface.parseIsolationLevel(transaction.options.isolationLevel),
+      );
+    }));
+  }
+
+  async commitTransaction(transaction: Transaction, _options: CommitTransactionOptions): Promise<void> {
+    if (!(transaction instanceof Transaction)) {
+      throw new TypeError('Unable to commit a transaction without transaction object!');
+    }
+
+    const connection = transaction.getConnection() as MsSqlConnection;
+    await connection.queue.enqueue(async () => new Promise<void>((resolve, reject) => {
+      connection.commitTransaction(error => (error ? reject(error) : resolve()));
+    }));
+  }
+
+  async rollbackSavepoint(transaction: Transaction, options: RollbackSavepointOptions): Promise<void> {
+    if (!(transaction instanceof Transaction)) {
+      throw new TypeError('Unable to rollback a transaction without transaction object!');
+    }
+
+    const connection = transaction.getConnection() as MsSqlConnection;
+    await connection.queue.enqueue(async () => new Promise<void>((resolve, reject) => {
+      // @ts-expect-error -- TODO: remove this when tedious types are fixed
+      connection.rollbackTransaction(error => (error ? reject(error) : resolve()), options.savepointName);
+    }));
+  }
+
+  async rollbackTransaction(transaction: Transaction, options: RollbackTransactionOptions): Promise<void> {
+    if (!(transaction instanceof Transaction)) {
+      throw new TypeError('Unable to rollback a transaction without transaction object!');
+    }
+
+    const connection = transaction.getConnection() as MsSqlConnection;
+    await connection.queue.enqueue(async () => new Promise<void>((resolve, reject) => {
+      // @ts-expect-error -- TODO: remove this when tedious types are fixed
+      connection.rollbackTransaction(error => (error ? reject(error) : resolve()), options.transactionName);
+    }));
+  }
+
+  async setIsolationLevel(_transaction: Transaction, _options: SetIsolationLevelOptions): Promise<void> {}
```

Comment:
I think this should throw and only be called by dialects that support changing the isolation level after the transaction has been started

For the rest, I think `startTransaction` should accept a `isolationLevel` option and handle it itself, instead of having `Transaction#begin` change the call order based on `settingIsolationLevelDuringTransaction`

By doing this, it becomes possible to expose a `setIsolationLevel` on `Transaction` for dialects that support changing it

Reply:
Made some changes to implement this, LMK what you think.

Reply:
Sure, I've moved all the set isolation level logic to `startTransaction`